testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -5,6 +5,13 @@
 # Texts and links
 #
 
+def wait_for_spa_transition(text)
+  repeat_until_timeout(timeout: 500, message: "Error on navigating to '#{text}'") do
```

Comment:
Please reduce the timeout, we don't wait 500 seconds to raise a failure if for some reason we have several of those failing the test suite will take ages to finish. Set something more reasonable, something that a real user can wait before try top tap somewhere in the screen :) what about 10 secs?